src/services/codefixes/convertToAsyncFunction.ts

```diff
@@ -0,0 +1,549 @@
+                        }
+                        return createVariableDeclarationOrAssignment(prevArgName!, getSynthesizedDeepClone(funcBody) as Expression, transformer);
+                    }
+                    else {
+                        return createNodeArray([createReturn(getSynthesizedDeepClone(funcBody) as Expression)]);
+                    }
+                }
+                break;
+        }
+        return createNodeArray([]);
+    }
+
+    function getReturnStatementsWithPromiseHandlersIndices(block: Block): number[] {
+        const indices: number[] = [];
+        for (let i = 0; i < block.statements.length; i++) {
+            const statement = block.statements[i];
+            if (getReturnStatementsWithPromiseHandlers(statement).length) {
+                indices.push(i);
+            }
+        }
+        return indices;
+    }
+
+    function removeReturns(stmts: NodeArray<Statement>, prevArgName: Identifier, constIdentifiers: Identifier[]): NodeArray<Statement> {
+        const ret: Statement[] = [];
+        for (const stmt of stmts) {
+            if (isReturnStatement(stmt)) {
+                if (stmt.expression) {
+                    ret.push(createVariableStatement(/*modifiers*/ undefined,
+                        (createVariableDeclarationList([createVariableDeclaration(prevArgName, /*type*/ undefined, stmt.expression)], getFlagOfIdentifier(prevArgName, constIdentifiers)))));
+                }
+            }
+            else {
+                ret.push(getSynthesizedDeepClone(stmt));
+            }
+        }
+
+        return createNodeArray(ret);
+    }
+
+
+    function getInnerTransformationBody(transformer: Transformer, innerRetStmts: Node[], prevArgName?: SynthIdentifier) {
+
+        let innerCbBody: Statement[] = [];
+        for (const stmt of innerRetStmts) {
+            forEachChild(stmt, function visit(node: Node) {
+                if (isCallExpression(node)) {
+                    const temp = transformExpression(node, transformer, node, prevArgName);
+                    innerCbBody = innerCbBody.concat(temp);
+                    if (innerCbBody.length > 0) {
+                        return;
+                    }
+                }
+                else if (!isFunctionLike(node)) {
+                    forEachChild(node, visit);
+                }
+            });
+        }
+        return innerCbBody;
+    }
+
+    function hasPropertyAccessExpressionWithName(node: CallExpression, funcName: string): boolean {
+        if (!isPropertyAccessExpression(node.expression)) {
+            return false;
+        }
+
+        return node.expression.name.text === funcName;
+    }
+
+    function getArgName(funcNode: Node, transformer: Transformer): SynthIdentifier {
+
+        const numberOfAssignmentsOriginal = 0, numberOfAssignmentsSynthesized = 0;
+        const types: Type[] = [];
+
+        function getMapEntryIfExists(node: Identifier): SynthIdentifier {

```

Comment:
Please have helper functions entirely at the bottom of their containing functions - it's confusing to have imperative lines both above and below them